app/models/debts/base_debt.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module Debts
+  # don't use this class directly, use OrganizationDebt or CustomerDebt
+  class BaseDebt
+    # basicaly instance can be organization or customer instance
+    def initialize(instance)
+      @instance = instance
+    end
+
+    def by_currency
+      invoices_group_by_currency.map do |currency, amount_cents|
+        Money.new(amount_cents, currency)
+      end
+    end
+
+    def total
+      def_curr = @organization.default_currency
```

Comment:
@dissident pass `@organization` through initializer or use "virtual" method. Don't use class variable which are defined in child classes.